godtools/App/Features/Tools/SwiftUI/AllTools/AllToolsContentView.swift

```diff
@@ -53,25 +29,29 @@ struct AllToolsContentView: View {
                 
             } else {
                 
-                GeometryReader { geo in
-                    let width = geo.size.width
+                GeometryReader { geometry in
+                    
+                    let geometryWidth: CGFloat = geometry.size.width
+                    let toolsVerticalSpacing: CGFloat = 10
+                    let toolsPaddingMultiplier: CGFloat = 20/375
+                    let leadingTrailingPadding: CGFloat = geometryWidth * toolsPaddingMultiplier
+                    let cardWidth: CGFloat = geometryWidth - (2 * leadingTrailingPadding)
                     
-                    if #available(iOS 15.0, *) {
+                    ItemsList<ResourceModel, ToolCardView>(hidesSeparatorInContainerTypes: [UIHostingController<AllToolsContentView>.self], listItems: $viewModel.tools, refreshList: {
+                        
+                        viewModel.refreshTools()
+                        
+                    }, viewForItem: { item in
                         
-                        // Pull to refresh is supported only in iOS 15+
-                        AllToolsList(viewModel: viewModel, width: width)
-                            .refreshable {
-                                viewModel.refreshTools()
-                            }
+                        let cardViewModel = viewModel.cardViewModel(for: item)
                         
-                    } else if #available(iOS 14.0, *) {
+                        return ToolCardView(viewModel: cardViewModel, cardWidth: cardWidth)
```

Comment:
Would it be possible to add the tap gesture here, before returning the view, rather than doing it inside `ItemsList`?

Reply:
Hey @rachaelblue I see what you're saying with it being difficult to build on this component.  There are ways that can be achieved.   The main thing I'm looking for is that code isn't being duplicated.  I'd love to have just the AllToolsContentView referencing the viewModel and any logic working with the viewModel is done in the AllToolsContentView and is done once, not repeated in multiple places.  Same with adding categories and spotlight, that code should be done once and not duplicated.  However that can be achieved is what is preferable. 
I'm definitely in favor of dropping List as well and sticking with the LazyVStack going forward (I haven't looked into pull to refresh on LazyVStack, but that option would be nice to have).  It would be nice if we could just go forward with iOS 14.